shape-blog-react/static/app/app.jsx

```diff
@@ -0,0 +1,33 @@
+import React, {Component} from 'react';
+import {hot} from 'react-hot-loader';
+import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
+import Header from './header';
+import Footer from './footer';
+import Homepage from './homepage';
+import AllPosts from './all-posts';
+import Post from './post';
+import Author from './author';
+import theme from './base.css';
+
+class App extends Component {
+  render() {
+    return (
+      <Router>
+        <div className={theme.page}>
+          <Header/>
+          <div className={theme.main}>
+            <Switch>
+              <Route exact path={'/'} component={Homepage}/>
```

Comment:
We should read this out of `tsg.yml` and then configure a mapping of route name to component.